internal/pkg/source/source.go

```diff
@@ -114,51 +115,49 @@ func sourcesFromClosures(fn *ssa.Function, conf *config.Config, taggedFields fie
 }
 
 // sourcesFromBlocks finds Source values created by instructions within a function's body.
-func sourcesFromBlocks(fn *ssa.Function, conf *config.Config, taggedFields fieldtags.ResultType) []*Source {
+func sourcesFromBlocks(fn *ssa.Function, conf *config.Config, taggedFields fieldtags.ResultType, propagators fieldpropagator.ResultType) []*Source {
 	var sources []*Source
 	for _, b := range fn.Blocks {
 		for _, instr := range b.Instrs {
-			// This type switch is used to catch instructions that could produce sources.
-			// All instructions that do not match one of the cases will hit the "default"
-			// and they will not be examined any further.
+			var isSource bool
```

Comment:
If you think it reads better, this entire switch could now be extracted to a `func isSourceInstruction(...)`, which would collapse this loop to something very succinct.

Reply:
I do think that would read better. If you want to extract that function, go ahead.

I think the code was mostly fine, I'm not sure about this `isSource` variable. You're calling `IsSourceType` in many places now, instead of just one earlier. This makes it harder to tell, at a glance, what the exclusions are for a given instruction type, because they all have this call. If you extract a function though you may be able to use early returns to get rid of that `isSource` variable.

Reply:
For a future PR, but once the body is a separate method, it'll just read 

```golang
for _, b := range fn.Blocks {
  for _, instr := range b.Instrs {
    if isSourceInstruction(...) {
      sources = append(sources, instr)
    }
  }
}
```

So the variable name is dropped altogether.

Of course, between now and the actual PR, I'll need to come up with a better name for the method...  it's naming all the way down, huh?